src/Omnipay/Common/HasParametersTrait.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace Omnipay\Common;
+
+trait HasParametersTrait
+{
+    /**
+     * Internal storage of all of the parameters.
+     *
+     * @var ParameterBag
+     */
+    protected $parameters;
```

Comment:
I have maybe missing something but `$parameters` is never used outside `HasParametersTrait`, no ? If it's the case, it should be `private` IMO.